src/main/java/useCase/GenerateUserHistory.java

```diff
@@ -88,23 +167,56 @@ public String getLikedLocation(UserAccount user){
         return commonLocation;
     }
 
-    // Use user history data to provide statistics on what kind of people the user likes by looking
-    // at the gender of the users that the user liked.
-    // The gender identified by user.
-    // 'M': male
-    // 'F': female
-    // 'N': non-binary
-    public String getLikedType(UserAccount user) {
-        String[] userCategories = new String[]{"Likes male users", "Likes female users",
-                "Likes non-binary users"};
+    /**
+     * Function that returns the most common country among the liked users
+     * @param user UserAccount of current user
+     * @return return a String of the most common country
+     */
+    public String getLikedCountry(UserAccount user){
+        String commonLocation = "";
+        String[] countries = new String[user.getLikedUsers().size()];
+
+        for (int i = 0; i < user.getLikedUsers().size(); i++) {
+            countries[i] = user.getLikedUsers().get(i).getLocation().get("country");
+        }
+
+        boolean[] alreadySeen = new boolean[countries.length];
+
+        int resultCount = 0;
+        for (int i = 0; i < countries.length; i++) {
+            if (!alreadySeen[i]) {
+                alreadySeen[i] = true;
+                int count = 1;
+                for (int k = i + 1; k < countries.length; k++) {
+                    if (!alreadySeen[k]) {
```

Comment:
You might want to put this into another method with extract method because the indentation and the lack of comments make it hard to understand. That being said, a bit of comments to explain what each loop does (can put it just before the loop starts) would help a lot.